File: app/src/main/java/com/sjianjun/reader/http/WebViewClient.kt
package com.sjianjun.reader.http

import android.os.Looper
import io.legado.app.help.http.BackstageWebView
import kotlinx.coroutines.DelicateCoroutinesApi
import kotlinx.coroutines.GlobalScope
import kotlinx.coroutines.launch
import sjj.alog.Log

//StringEscapeUtils.unescapeJson
object WebViewClient {
    @OptIn(DelicateCoroutinesApi::class)
    @JvmStatic
    fun get(
        url: String? = null,
        headers: Map<String, String>? = null,
        javaScript: String = "document.documentElement.outerHTML",
        timeout: Long = 20000L,
    ): String {
        if (Looper.myLooper() == Looper.getMainLooper()) {
            throw IllegalStateException("WebViewClient 必须在子线程中调用")
        }
        Log.i("web_get url: $url, headerMap: $headers, javaScript: $javaScript, timeout: $timeout")
        var result: String? = null
        GlobalScope.launch {
            try {
                result = BackstageWebView(
                    url = url,
                    headerMap = headers,
                    javaScript = javaScript,
                    timeout = timeout
                ).getResponse()
            } catch (e: Exception) {
                Log.e("WebViewClient get error", e)
                result = ""
            }
        }
        // 等待协程执行完成
        while (result == null) {
            Thread.sleep(100)
        }
        // 处理返回结果

        return result
    }
}

File: app/src/main/python/WebViewClient.py
from java.util import HashMap
from com.sjianjun.reader.http import CookieMgr
from com.sjianjun.reader.http import WebViewClient

def web_get(url,headers=None, javaScript="document.documentElement.outerHTML", timeout=20000):
    if headers is None:
        headers = {}
    if javaScript is None:
        javaScript = "document.documentElement.outerHTML"
    # 将 Python 的 dict 转换为 Java 的 HashMap
    java_headers = HashMap()
    for key, value in headers.items():
        java_headers.put(key, value)
    return WebViewClient.get(url, java_headers, javaScript,timeout)
